reference/conanfile/attributes.rst

```diff
@@ -1028,6 +1028,74 @@ has **the same default directories**.
 Dependencies among components and to components of other requirements can be defined using the ``requires`` attribute and the name
 of the component. The dependency graph for components will be calculated and values will be aggregated in the correct order for each field.
 
+There is a new way of setting and accessing ``filenames``, ``names`` and ``build_modules`` starting
+in Conan 1.36 using new ``set_property`` and ``get_property`` methods of the ``cpp_info`` object:
+
+.. code-block:: python
+
+    def set_property(self, property_name, value, generator=None)
+    def get_property(self, property_name, generator=None):
+
+New properties ``cmake_target_name``, ``cmake_file_name``, ``pkg_config_name`` and
+``cmake_build_modules`` are defined to allow migrating ``names``, ``filenames`` and ``build_modules``
+properties to this model.
+In Conan 2.0 this will be the default way of setting these properties and also passing custom
+properties to generators. 
+
```

Comment:
```suggestion
In Conan 2.0 this will be the default way of setting these properties and also passing custom
properties to generators. 

For most cases, it is recommended not to use the ``generator`` argument. The properties are generic for build systems, and different generators that integrate with a given build system could be reading such generic properties.
```